### PR TITLE
fix property_set when depth > 0

### PR DESCRIPTION
**Pull request by ddelbondio**
I am not 100% sure about the EG(return_value_ptr_ptr) restoring, but otherwise this seems to work.

### CODE REVIEW

File: xdebug_handler_dbgp.c [hunk 1]
```diff
@@ -1898,10 +1900,39 @@ DBGP_FUNC(property_set)
...
 		}
 	} else {
+		/* backup executor state */
+		if(depth > 0) {
+			original_execute_data = EG(current_execute_data);
+			original_return_value = EG(return_value_ptr_ptr);
+
+			EG(current_execute_data) = XG(active_execute_data);
+
+			EG(opline_ptr) = &EG(current_execute_data)->opline;
```
Comment (derickr) [inline xdebug_handler_dbgp.c]: xdebug_do_eval already handles many of those. Do you have a very specific reason as why to store them again?

Comment (ddelbondio) [inline xdebug_handler_dbgp.c]: xdebug_do_eval only backs up the current values, but i am actually setting all the EG() values to the appropriate values of the stackframe indicated by depth.

Comment (derickr) [inline xdebug_handler_dbgp.c]: I've just tried, and it doesn't seem to have any effect at all :-)

**Comment by derickr** [comment]
I've merged this. Thanks for the patch!

**Pull request closed without merging**